source/common/ur_singleton.h

```diff
@@ -0,0 +1,73 @@
+/*
+ *
+ * Copyright (C) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+#pragma once
```

Comment:
Although this is probably supported by almost any compiler, I think it is still non-standard, so it might be better to favour a traditional `#ifndef` guard.